pkg/hubble/monitor/consumer.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitor
+
+import (
+	"time"
+
+	"github.com/cilium/cilium/pkg/hubble/observer"
+	observerTypes "github.com/cilium/cilium/pkg/hubble/observer/types"
+	"github.com/cilium/cilium/pkg/monitor/agent/consumer"
+	nodeTypes "github.com/cilium/cilium/pkg/node/types"
+)
+
+type hubbleConsumer struct {
+	observer      observer.GRPCServer
+	numEventsLost uint64
+}
+
+// NewHubbleConsumer returns an initialized pointer to hubbleConsumer.
+func NewHubbleConsumer(observer observer.GRPCServer) consumer.MonitorConsumer {
+	mc := &hubbleConsumer{
+		observer:      observer,
+		numEventsLost: 0,
+	}
+	return mc
+}
+
+func (mc *hubbleConsumer) sendEventQueueLostEvents() {
+	lostEvent := &observerTypes.MonitorEvent{
+		Timestamp: time.Now(),
+		NodeName:  nodeTypes.GetName(),
+		Payload: &observerTypes.LostEvent{
+			Source:        observerTypes.LostEventSourceEventsQueue,
+			NumLostEvents: mc.numEventsLost,
+		},
+	}
+	select {
+	case mc.observer.GetEventsChannel() <- lostEvent:
+		mc.numEventsLost = 0
```

Comment:
Why is this reset to `0`? Shouldn't it be decremented by `1` instead?

Reply:
So the overall idea here is that we inform Hubble about the number of events that got dropped in this consumer. So Hubble will receive an event with `4 evens got dropped` if `numEventsLost`  was 4.

Therefore, once we successfully sent the event, we can clear the counter by setting it to 0, as all lost events have been submitted.

Reply:
I see. I think the naming for `lostEvent` is a bit confusing then as it reads like if you're sending a previously lost event to the events channel. It's probably also worth adding a comment here I think.

Reply:
I have added comments and also slightly changed some variable names. Hope this makes the intent more clear.

Reply:
Yup, the intent is very clear now. Thanks!